### PR TITLE
Implements connectToStores()

### PR DESCRIPTION
**Pull request by cpsubrian**
First stab at the 'higher order component' discussed in #52. I'm open to feedback on any language or implementation details. I'm pretty new to React and especially new to testing React code, so let me know how everything looks

- Tests (with new ones) all pass
- Linter passes
- Coverage is **NOT at 100%**
  - There is some transpiled code that is not reached and I'm not sure of the best way to handle that. 
  - See screenshot below.

Note: I had to rework the `ReactComponent` helper a bit to facilitate my tests. I'm pretty sure I fixed a typo/error with the nesting of the static property copying inside the mixin copying.

### CODE REVIEW

**Comment by cpsubrian** [comment]
Coverage:

screenshot 2015-03-30 13 09 39

File: utils/connectToStores.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    componentWillUnmount () {
+      stores.forEach(store => store.unlisten(this.onChange))
+    },
+
+    onChange () {
+      this.setState(Component.getPropsFromStores(this.props))
+    },
+
+    render () {
+      return React.createElement(Component, Object.assign({}, this.props, this.state))
```
Comment (cpsubrian) [inline utils/connectToStores.js]: Noticed you're not using JSX in most places, but we could replace this with (what I use):

```jsx
render() {
  return <Component {...this.props} {...this.state} />
}
```

Comment (troutowicz) [inline utils/connectToStores.js]: This is personal preference of course, but I tend to not use JSX in utilities, only in components. Makes it easier to understand the purpose of the file for me atleast.

Comment (cpsubrian) [inline utils/connectToStores.js]: Yep, not big deal to me either way which is why I tried to stick to the common style.

Comment (cpsubrian) [inline utils/connectToStores.js]: I'm gonna go out on a limb and guess that `Object.assign` is very similar to however spread attributes are implemented anyway :)

Comment (troutowicz) [inline utils/connectToStores.js]: It's what I used when I modified your util for a personal project yesterday. :) It just returns a new object with the properties and state together.

**Comment by cpsubrian** [comment]
Pushed a commit that should fix the lint error.

**Comment by troutowicz** [comment]
What do you guys think about including bootstrap ability in this util.

```javascript
const alt = require('../alt');

...

componentWillMount() {
  if (this.props.altStores) {
    alt.bootstrap(JSON.stringify(this.props.altStores));

    // force state update this render cycle
    this.setState(Component.getStateFromStores(this.props));
  }
},

...
```

**Comment by cpsubrian** [comment]
Looks like I need to read-up/try-out bootstrapping :)

**Comment by troutowicz** [comment]
It's useful with an isomorphic app and passing initial data from server to clients.

File: utils/connectToStores.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
+'use strict'
+import React from 'react'
```
Comment (goatslacker) [inline utils/connectToStores.js]: This is causing React to be included in the alt-browser bundle!!!! :astonished:

Comment (cpsubrian) [inline utils/connectToStores.js]: Should we just have the user pass in React?

Comment (goatslacker) [inline utils/connectToStores.js]: Probably assume people have react (window.React) in the browser version.

Comment (cpsubrian) [inline utils/connectToStores.js]: 09b64ec removes React from the build, but not sure if `window.React` gets picked up. Need to work up a client-side test I guess.

**Comment by cpsubrian** [comment]
ee2c04a makes some annoyingly trivial syntax changes such that the coverage is back at 100%. Guess we can't have all the ES6 without specific tests for the shims :)

**Comment by troutowicz** [comment]
I'll post my modified version here in case it adds something worth while.

[connectToStores.js](https://github.com/troutowicz/geoshare/blob/master/app/utils/connectToStores.js)

**Comment by cpsubrian** [comment]
@troutowicz Thanks! I intend to look into the bootstrapping with I get a chance.

Curious why you prefer the name `getStateFromStores` vs. `getPropsFromStores`? In the context of where the method is defined you are always 'updating' the props of the wrapped Component.

**Comment by troutowicz** [comment]
We are always passing in data via props, but the data itself is coming from a store. Stores contain state. Either way would work though.

**Comment by goatslacker** [comment]
I've rebased your commits into #177 lets get this in :dart:

**Pull request closed without merging**